synapse/storage/state.py

```diff
@@ -353,8 +353,158 @@ def get_member_split(self):
         return member_filter, non_member_filter
 
 
+class StateGroupBackgroundUpdateStore(SQLBaseStore):
+    """Defines functions related to state groups needed to run the state backgroud
+    updates.
+    """
+
+    def _count_state_group_hops_txn(self, txn, state_group):
```

Comment:
So we're putting some functions other than background updates in the bg update store? That's OK I guess but that doesn't really fit with the fact that registration store got split into three

Reply:
In this case, `_count_state_group_hops_txn` is called from one of the background updates.